client/render-system.cpp

```diff
@@ -146,32 +100,40 @@ void RenderSystem::Startup() {
 	// Black is the safest clear color since this is a space game.
 	glClearColor(0.0, 0.0, 0.0, 0.0);
 	// Reversed Z buffering for improved precision (maybe)
-	glClearDepth(0.0f);
+	glClearDepth(0.0);
 	glDepthFunc(GL_GREATER);
-	std::shared_ptr<OBJ> sphere = OBJ::Create(Path("assets:/sphere/sphere.obj"));
-	if (!sphere) {
-		_log->debug("[RenderSystem] Error loading sphere.obj.");
-	}
-	else {
-		this->sphere_vbo.Load(sphere);
-	}
-	std::shared_ptr<OBJ> quad = OBJ::Create(Path("assets:/quad/quad.obj"));
-	if (!quad) {
+
+	if (const auto quad = OBJ::Create(Path("assets:/quad/quad.obj")); !quad) {
 		_log->debug("[RenderSystem] Error loading quad.obj.");
 	}
 	else {
-		this->quad_vbo.Load(quad);
+		quad_vbo = std::make_unique<VertexBufferObject>(vertex::VF_BASE);
+		quad_vbo->Load(quad);
+	}
+
+	this->render_passes.push_back(std::make_shared<graphics::pass::GeometryPass>());
+	this->render_passes.push_back(std::make_shared<graphics::pass::PointLightPass>());
+	this->render_passes.push_back(std::make_shared<graphics::pass::DirLightPass>());
+	this->render_passes.push_back(std::make_shared<graphics::pass::FinalPass>());
+
+	if (options.debug_gbuffer()) {
+		this->render_passes.push_back(std::make_shared<graphics::pass::DebugPass>());
 	}
 
-	this->inv_view_size.x = 1.0f / static_cast<float>(this->view_size.x);
-	this->inv_view_size.y = 1.0f / static_cast<float>(this->view_size.y);
-	this->light_gbuffer.AddColorAttachments(this->view_size.x, this->view_size.y);
-	this->light_gbuffer.SetDepthAttachment(GBuffer::DEPTH_TYPE::DEPTH, this->view_size.x, this->view_size.y);
+	glm::uvec2 view_size{1024, 768};
```

Comment:
I know this variable was probably copied from the old version of the header, but... `default-config.hpp` is a thing with nice constants... ***however***... `SetViewportSize` is called almost right after `Startup`, so setting the viewport here at all seems a little moot, if Adding the Color/Depth attachments were deferred to the first call of `SetViewportSize`, then that also wouldn't even need to be here.

Reply:
I was planning this change in another PR, figured it was best to reduce overall changes within this already big PR.

Reply:
I did some of this work by creating a helper to update the viewport instance. I added calls to this methods in both the Startup and SetViewportSize methods, so they are both calling the same source of truth for updating the viewport.